### DISCUSSION TITLE
Issues with Splines: Bug or Feature?

### DISCUSSION THREAD

**Discussion by dapa5900** [JacquesLucke/animation_nodes#1681] [Q&A] [answered]
Hi @OmarEmaraDev,

as suggested by you yesterday I would like to ask here regarding two issues that I encountered while playing around with the spline nodes where I don't know if this is a bug or intended behavior (and if so, how can it be avoided?):

Please fine here the file for reference: https://p7s1-my.sharepoint.com/:f:/g/personal/par0001d_belgium_fhm_de/EvfrYas9yktOu3IJpphYWfMBjtxgNFTDMVHHvnuz0nEHhQ?e=RrsEXB

1. As you can see I created a a procedural spline utilizing the distribute matrices node and setting it to Cyclic in the Spline from Points node. It works as expected however if I want to tilt the spline in Accumulate mode it doesn't acknowledge the whole cyclic spline but repeats the tilt for the last segment.

2. In the folder I've shared is also a short video showcasing the other issue where I can not set the shading mode to Flat permanently since whenever I change something on the nodes, the shading get's set back to smooth.

Thank you!

**Comment by OmarEmaraDev** [comment, marked as answer]
1. Tilt is stored per bezier point, so in order to get the tilt at any point on the spline segment, you need to interpolate the tilt of the two bezier points it lies between. The problem is for the last segment, the two bezier points are the first and last points of the spline. In accumulate mode, the first point will have a small tilt value while the last point will have a significantly higher value, so interpolating the two will yield a rapidly changing tilt, causing the the effect you see. I am not sure how this can be avoided. Adding an option to instead assign a lower equivalent angle tilt might be a good solution, but I will have to check how Blender treats equivalent angles in this case.
2. It seems each spline in Blender defines a Use Smooth value that will smooth that particular spline. This is initialized to true by default, hence the result you see. I see two possible solutions, either add that to the spline (Like Use Cyclic) or add an input for it in the Curve Object Output node. I am likely to take the latter approach as the option will nearly never be used as far as I can tell.

I will pick this up again in a few days.

**Reply by OmarEmaraDev** [reply]
@dapa5900  You can now use the Shade Object Smooth node to change the smoothness of the curve. 765ad30

For tilt interpolation, I tried couple of things, but non of them worked. So I would just call this an expected behavior. To work around this, you can generate a non cyclic spline that matches the cyclic one visually and tilt it as needed.

**Reply by dapa5900** [reply]
Great thank you
